packages/rocketchat-lib/server/functions/deleteMessage.js

```diff
@@ -15,12 +16,17 @@ RocketChat.deleteMessage = function(message, user) {
 		}
 	} else {
 		if (!showDeletedStatus) {
+			deletedMsg = RocketChat.models.Messages.findOneById(message._id);
 			RocketChat.models.Messages.removeById(message._id);
 		}
 
 		if (message.file && message.file._id) {
 			FileUpload.delete(message.file._id);
 		}
+
+		Meteor.defer(function() {
+			RocketChat.callbacks.run('afterDeleteMessage', deletedMsg);
```

Comment:
New event so interested modules can process/cleanup based on the message being deleted.  In my case, post a msg to slack to have them delete the message as well.